### PR TITLE
Migrate network configuration safely

### PR DESCRIPTION
**Pull request by cvium**
Supersedes #5482

### CODE REVIEW

File: Jellyfin.Server/Migrations/PreStartupRoutines/CreateNetworkConfiguration.cs [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+        public int HttpsPortNumber { get; set; } = DefaultHttpsPort;
+
+        public bool EnableHttps { get; set; }
+
+        public int PublicPort { get; set; } = DefaultHttpPort;
+
+        public bool EnableIPV6 { get; set; }
+
+        [DataMember(Name = "EnableIPV4")]
```
Comment (Bond-009) [inline Jellyfin.Server/Migrations/PreStartupRoutines/CreateNetworkConfiguration.cs:114]: Why is this needed?

File: Jellyfin.Server/Migrations/PreStartupRoutines/CreateNetworkConfiguration.cs [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+        using var xmlWriter = XmlWriter.Create(path, xmlWriterSettings);
+        networkConfigSerializer.Serialize(xmlWriter, networkSettings);
+    }
+
+#pragma warning disable CS1591
+    public sealed class OldNetworkConfiguration
+    {
+        public const int DefaultHttpPort = 8096;
+
+        public const int DefaultHttpsPort = 8920;
```
Comment (Bond-009) [inline Jellyfin.Server/Migrations/PreStartupRoutines/CreateNetworkConfiguration.cs:62]: What's the point of these?

**Review by Bond-009 — CHANGES_REQUESTED** [review]

File: Jellyfin.Server/Migrations/PreStartupRoutines/CreateNetworkConfiguration.cs (hunk 1 above)
Comment (cvium) [reply Jellyfin.Server/Migrations/PreStartupRoutines/CreateNetworkConfiguration.cs:114]: It's not. It was from a previous iteration.

File: Jellyfin.Server/Migrations/PreStartupRoutines/CreateNetworkConfiguration.cs (hunk 2 above)
Comment (cvium) [reply Jellyfin.Server/Migrations/PreStartupRoutines/CreateNetworkConfiguration.cs:62]: They are referenced in multiple places?

File: Jellyfin.Server/Migrations/MigrationRunner.cs [hunk 3]
```diff
@@ -29,6 +42,7 @@ public sealed class MigrationRunner
             typeof(Routines.MigrateAuthenticationDb)
         };
 
+
```
Comment (crobibero) [inline Jellyfin.Server/Migrations/MigrationRunner.cs:45]:
```suggestion
```

**Review by crobibero — APPROVED** [review]

**Review by Bond-009 — APPROVED** [review]
